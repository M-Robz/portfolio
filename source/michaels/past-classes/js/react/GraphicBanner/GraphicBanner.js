import React from 'react';

/**
 * A banner with copy in the image.
 * @param {Object}  props
 * @param {string}  props.srcMobile - Mobile image source.
 * @param {string}  props.srcDesktop - Desktop image source.
 * @param {string}  props.readerMarkup - Markup visible only to screen readers.
 * @param {string}  [props.href] - The link destination.
 * @param {boolean} [props.outlined] - Whether the banner is outlined.
 */
export default function GraphicBanner({
    srcMobile, srcDesktop, readerMarkup, href, outlined
}) {

    const image = (
        <picture>
            <source srcSet={srcMobile} media="(max-width: 480px)" />
            <img className="c-banner__img" src={srcDesktop} alt="" />
        </picture>
    );

    const copy = (
        <div className="c-banner__copy sr-only" dangerouslySetInnerHTML={
            { __html: readerMarkup }
        }></div>
    );

    return (
        <div className={`c-banner u-mb-1r${ outlined ? ' u-outline' : '' }`}>
            { href ? (
                <a className="c-banner__link" href={href}>
                    {image}
                    {copy}
                </a>
            ) : (
                <>
                    {image}
                    {copy}
                </>
            ) }
        </div>
    );
}
