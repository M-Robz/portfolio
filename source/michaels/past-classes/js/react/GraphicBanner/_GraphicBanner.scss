// Dependencies
//   - Mixins: `set-aspect-ratio-container`, `set-aspect-ratio-before`

.c-banner {
    @include set-aspect-ratio-container(1280, 315);

    @media (max-width: 480px) {
        // Most category-specific banners have a 480x550 aspect ratio, but the
        // generic banner that's currently in use is 325x438
        @include set-aspect-ratio-before(325, 438);
    }
}
