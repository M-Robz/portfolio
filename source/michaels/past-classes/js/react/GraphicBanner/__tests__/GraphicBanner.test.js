import React from 'react';
import renderer from 'react-test-renderer';
import GraphicBanner from '../GraphicBanner';

const REQUIRED_PROPS = {
    srcMobile: 'mobile-image.jpg',
    srcDesktop: 'desktop-image.jpg',
    readerMarkup: '<h2>Lorem ipsum</h2>'
};

const OPTIONAL_PROPS = {
    href: 'test/link',
    outlined: true
};

it('renders correctly with all props', () => {
    const tree = renderer
        .create(<GraphicBanner {...REQUIRED_PROPS} {...OPTIONAL_PROPS} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with only required props', () => {
    const tree = renderer
        .create(<GraphicBanner {...REQUIRED_PROPS} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
