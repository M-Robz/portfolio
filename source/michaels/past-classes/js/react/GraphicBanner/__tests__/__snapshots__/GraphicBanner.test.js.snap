// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<div
  className="c-banner u-mb-1r u-outline"
>
  <a
    className="c-banner__link"
    href="test/link"
  >
    <picture>
      <source
        media="(max-width: 480px)"
        srcSet="mobile-image.jpg"
      />
      <img
        alt=""
        className="c-banner__img"
        src="desktop-image.jpg"
      />
    </picture>
    <div
      className="c-banner__copy sr-only"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2>Lorem ipsum</h2>",
        }
      }
    />
  </a>
</div>
`;

exports[`renders correctly with only required props 1`] = `
<div
  className="c-banner u-mb-1r"
>
  <picture>
    <source
      media="(max-width: 480px)"
      srcSet="mobile-image.jpg"
    />
    <img
      alt=""
      className="c-banner__img"
      src="desktop-image.jpg"
    />
  </picture>
  <div
    className="c-banner__copy sr-only"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h2>Lorem ipsum</h2>",
      }
    }
  />
</div>
`;
